### PR TITLE
Expose `schema.sdl` add some helper types and traits

### PR DESCRIPTION
**Pull request by Dentosal**
Work towards https://github.com/FuelLabs/fuel-subgraph/issues/11
Fixes todo pointed out here: https://github.com/FuelLabs/fuel-indexer/pull/969/files#r1214834302

### CODE REVIEW

File: crates/client/src/client/schema/block.rs [hunk 1]
```diff
@@ -84,6 +111,16 @@ pub struct Block {
     pub transactions: Vec<TransactionIdFragment>,
 }
 
+/// Block with full transaction data
+#[derive(cynic::QueryFragment, Debug)]
+#[cynic(schema_path = "./assets/schema.sdl", graphql_type = "Block")]
+pub struct FullBlock {
```
Comment (xgreenx) [inline crates/client/src/client/schema/block.rs:117]: Do we really want to include it into the main `fuel-core-client`? It is available via [`fuel-core-client-ext`](https://github.com/FuelLabs/fuel-core-client-ext) and not used by the DSK

Comment (Dentosal) [reply crates/client/src/client/schema/block.rs:117]: Oh I didn't realize the -ext was a thing. I'll remove these.

**Comment by Dentosal** [comment]
CI is hitting docker.com rate limits :/

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 
+### Added
+
+- [#1731](https://github.com/FuelLabs/fuel-core/pull/1731): Expose `schema.sdl` from `fuel-core-client`. Also adds more trait impls and a way to query full block info directly.
```
Comment (xgreenx) [inline CHANGELOG.md:11]:
```suggestion
- [#1731](https://github.com/FuelLabs/fuel-core/pull/1731): Expose `schema.sdl` from `fuel-core-client`.
```

**Pull request merged**